include/deal.II/matrix_free/matrix_free.h

```diff
@@ -2235,6 +2239,12 @@ MatrixFree<dim, Number, VectorizedArrayType>::initialize_dof_vector(
   VectorType &       vec,
   const unsigned int comp) const
 {
+  static_assert(IsBlockVector<VectorType>::value == false,
+                "This function is not supported for block vectors.");
+
+  Assert(task_info.n_procs == 1,
+         ExcMessage("This function can only be used in serial."));
```

Comment:
I also added here some asserts. Hope my understanding about `serial non-block vector` is correct.